### PR TITLE
Adding Gitlab configuration with proper test environment config

### PR DESCRIPTION
**Pull request by brianghig**
Fixing Mongoose tests to fail if the DB connection cannot initialize

### CODE REVIEW

**Review by ymao1 — APPROVED** [review]

**Review by jrassa — APPROVED** [review]
Looks good to me.  Lines up with what we are already doing on gitlab.

File: package.json [hunk 1]
```diff
@@ -16,8 +16,8 @@
     "start": "nodemon ./src/server.js --exec 'npm run lint && node --inspect' | bunyan",
     "start:dev": "export NODE_ENV=development && yarn start",
     "test": "NODE_ENV=test nodemon ./src/test.js",
-    "test:ci": "NODE_ENV=test nyc --reporter=lcov --reporter=html node ./src/test.js --ci",
-    "test:gitlab": "NODE_ENV=testgitlab nyc --reporter=lcov --reporter=html node ./src/test.js --ci",
+    "test:ci": "NODE_ENV=test nyc --reporter=lcov node ./src/test.js --ci",
```
Comment (brianghig) [inline package.json:19]: @jrassa I'd like to add the html reporter back in to this. It helps to visualize the coverage on a local dev environment without requiring any other infrastructure. Any objections? Thoughts?

Comment (jrassa) [reply package.json:19]: Sorry, I meant to explain that change.  The lcov report also generates the html report.  Including both causes it to generate 2 copies of the html report.

https://istanbul.js.org/docs/advanced/alternative-reporters/#lcov

Comment (brianghig) [reply package.json:19]: That's amazing! I've been overlooking that for quite a while. Thanks again @jrassa !

**Pull request merged**